kfp/website-ingestor/ingestion-pipeline-website-local.py

```diff
@@ -0,0 +1,446 @@
+            logger.error(f"Error creating index '{index_name}': {e}")
+
+    def convert_to_md(html_content, url):
+        # Convert HTML to Markdown using Html2Text
+        transformer = Html2TextTransformer()
+        document = Document(page_content=html_content, metadata={"url": url})
+        md_document = transformer.transform_documents([document])[0]
+
+        # Split Markdown into sections based on headers
+        headers_to_split_on = [
+            ("#", "Header1"),
+            ("##", "Header2"),
+            ("###", "Header3"),
+            ("####", "Header4"),
+        ]
+        markdown_splitter = MarkdownHeaderTextSplitter(headers_to_split_on=headers_to_split_on, strip_headers=True)
+        header_splits = markdown_splitter.split_text(md_document.page_content)
+
+        # Further split content into chunks for Weaviate ingestion
+        text_splitter = RecursiveCharacterTextSplitter(chunk_size=2048, chunk_overlap=256)
+        splits = text_splitter.split_documents(header_splits)
+
+        # Prepare for JSON format with content and metadata
+        json_splits = []
+        for split in splits:
+            content_header = f"Section: {split.metadata.get('Header1', '')}"
+            for header_name in ["Header2", "Header3", "Header4"]:
+                if header_name in split.metadata:
+                    content_header += f" / {split.metadata[header_name]}"
+            content_header += "\n\nContent:\n"
+            split.page_content = content_header + split.page_content
+            json_splits.append({"page_content": split.page_content, "metadata": split.metadata})
+
+        logger.info(f"Successfully processed and converted content for {url}")
+        return json_splits
+
+    # Function to ingest data to Weaviate
+    def ingest(index_name, splits, weaviate_client):
+        """Ingest documents into Weaviate."""
+        logger.info(f"Starting ingestion for index: {index_name}")
+        try:
+            #model_kwargs = {"trust_remote_code": True, "device": "cuda"}
+            model_kwargs = {"trust_remote_code": True}
+            embeddings = HuggingFaceEmbeddings(
+                model_name="nomic-ai/nomic-embed-text-v1",
+                model_kwargs=model_kwargs,
+                show_progress=True,
+            )
+
+            db = Weaviate(
+                embedding=embeddings,
+                client=weaviate_client,
+                index_name=index_name,
+                text_key="page_content",
+            )
+            db.add_documents(splits)
+            logger.info(f"Successfully uploaded documents to {index_name}")
+        except Exception as e:
+            logger.error(f"Error during ingestion for index {index_name}: {e}")
+
+
+    """Process the website content and add it to the Weaviate store."""
+    logger.info(f"Starting processing for URL: {url}")
+    weaviate_client = get_weaviate_client()
+
+
+    # Ensure the index (class) exists
+    index_name = "WebScrapedData"
```

Comment:
Can we make this dynamic and also something that gets passed into the pipeline by tekton

Reply:
its dynamic, we can pass website url in the pipeline run

Reply:
Added the vectordb index as a pipeline parameter